test/game5Test.js

```diff
@@ -6,16 +6,33 @@ describe('Game5', function () {
     const Game = await ethers.getContractFactory('Game5');
     const game = await Game.deploy();
 
-    return { game };
+    return { game};
   }
   it('should be a winner', async function () {
-    const { game } = await loadFixture(deployContractAndSetVariables);
+    const { game} = await loadFixture(deployContractAndSetVariables);
 
-    // good luck
+    // creating random wallet that its address matches the condition
+    const threshold = 0x00FfFFfFFFfFFFFFfFfFfffFFFfffFfFffFfFFFf
+    let validAddress = false;
+    let wallet;
+    let address;
+    while (!validAddress) {
+      wallet = ethers.Wallet.createRandom()
+      address = await wallet.getAddress()
+      if(address < threshold) validAddress = true
```

Comment:
Hope you don't mind me asking, but how does comparing a string _(`address`)_ with a number _(`threshold`)_ work here?

I was struggling with this test file as I couldn't figure out how to identify which address would be less than the `threshold` address; when I convert an address to a number _(either with `Number(address)` or `parseInt(address)`)_ I could find an address that was less than the `threshold` number, but they wouldn't work.
I'm guessing I don't really understand how Hexadecimal numbers work? 🤷

Reply:
@skplunkerin JS let you compare the numbers just like that, but a better way can be adding an "n" to the threshold, so JS knows this is a bigInt and can compare them more accurately

Reply:
aw, that makes sense. Thanks for replying! I forget that an address is a hexadecimal, and that hexadecimal's are numbers.

Reply:
how long the loop ran?

Reply:
I thought it was supposed to be ajoke?

Reply:
@apoorv-2204 I tried several times and it ran between 100-300 times, which took a couple of seconds.